src/WebJobs.Script.WebHost/Management/IMeshServiceClient.cs

```diff
@@ -16,5 +17,7 @@ public interface IMeshServiceClient
         Task MountFuse(string type, string filePath, string scriptPath);
 
         Task PublishContainerActivity(IEnumerable<ContainerFunctionExecutionActivity> activities);
+
+        Task NotifyHealthEvent(ContainerHealthEventType healthEventType, Type source, string details);
```

Comment:
I wonder if you shouldn't generalize this to PublishEvent so it can be used for other event types in the future? Then you'd have ContainerEventType, ContainerEvent, etc.

Reply:
I spent some time thinking about this but couldn't come up with other event types we want to publish :)
So leaning towards just leaving this as is for now and generalize the api later if required. 
Any other events are you thinking of? The primary purpose for publishing the health event is to take remedial action at runtime vs just for logging purposes.